sycl/doc/extensions/supported/sycl_ext_intel_queue_submission_mode.asciidoc

```diff
@@ -0,0 +1,106 @@
+= sycl_ext_intel_queue_submission_mode
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 7 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This extension is implemented and fully supported by {dpcpp}.
+
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices
+when using the Level Zero backend.
+
+== Overview
+
+This extension enables specifying one of two submission modes for a queue.
+
+The property `no_immediate_command_list` specifies that submissions to a SYCL
+queue should be collected into groups before actual submission to the hardware.
+
+The property `immediate_command_list` specifies that submissions to a
+SYCL queue should be submitted to the hardware immediately.
+
+When the submission mode is left unspecified a platform-specific default
+is chosen by the implementation.
+
+
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_QUEUE_SUBMISSION_MODE` to one of the values defined in
+the table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
```

Comment:
Add a section here describing the queue properties.  Include a code synopsis which shows the namespace that includes each property.  For example:

> === New queue properties
>
> This extension adds the following new properties that can be used when constructing a `queue` object.
>
> [synopsis]
>
> Both properties are hints, which are ignored unless the backend is Level Zero.
>
> The property `immediate_command_list` requests that the implementation use an immediate command list when commands are submitted to this queue.  As a result, these commands are submitted immediately to the device.
>
> The property `no_immediate_command_list` requests that the implementation use a standard command queue instead of an immediate command list.  As a result. commands submitted to this queue may be batched with other commands before being submitted to the device.
>
> These two properties are mutually exclusive.  Constructing a `queue` with both properties causes the constructor to throw a synchronous `exception` with the `errc::invalid` error code.

Reply:
Ok